js/src/dropdown.js

```diff
@@ -289,6 +289,31 @@ class Dropdown extends BaseComponent {
     return this._element.closest(`.${CLASS_NAME_NAVBAR}`) !== null
   }
 
+  _getOffset() {
+    let { offset } = this._config
+
+    if (!offset) {
+      return [0, 0]
+    }
+
+    if (typeof offset === 'number') {
+      offset = [offset, 0]
+    } else if (typeof offset === 'string') {
+      offset = offset.split(',')
+      if (offset.length === 1) {
+        offset = [offset[0], 0]
+      }
```

Comment:
IMO, now configuration should be updated as per the Popper. So it is better to accept `OffsetsFunction | [?number, ?number]`. But you can keep the string type so that offset can be passed from data attributes.

And as per your check:
```js
if (offset.length === 1) {
  offset = [offset[0], 0]
}
``` 
You are allowing and set only **Skidding** , because `[number, number]` => [[Skidding](https://popper.js.org/docs/v2/modifiers/offset/#skidding-1), [Distance](https://popper.js.org/docs/v2/modifiers/offset/#distance-1)]. It may or may not be the desired behavior. So you should mention it in the docs.

Reply:
You're right, the configuration could be too much "smarty".
The offset option should be a shortcut.
So using javascript is better `OffsetsFunction | [?number, ?number]` as you said, 
with data attribute is better to use the `"Number,Number"` format.
If anybody needs more can use popperConfig :D